### PR TITLE
GetBenchmark stats now returns the snapshot

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

**Comment by zeptin** [comment]
@fassadlr Can you fix the conflict?

**Comment by noescape00** [comment]
@fassadlr please fix tests

**Comment by fassadlr** [comment]
@noescape00 fixed 👍

File: src/Stratis.Bitcoin/Utilities/Statistics/INodeBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Stratis.Bitcoin.Utilities.Statistics
+{
+    /// <summary>
+    /// Used to display benchmark information about a particular node component.
+    /// </summary>
+    public interface INodeBenchmark
```
Comment (noescape00) [inline src/Stratis.Bitcoin/Utilities/Statistics/INodeBenchmark.cs]: Why empty interface?

File: src/Stratis.Bitcoin/Utilities/NodeStats.cs [hunk 2]
```diff
@@ -35,6 +40,7 @@ public NodeStats(IDateTimeProvider dateTimeProvider)
             this.locker = new object();
             this.dateTimeProvider = dateTimeProvider;
 
+            this.nodeBenchmarks = new List<Func<INodeBenchmark>>();
```
Comment (noescape00) [inline src/Stratis.Bitcoin/Utilities/NodeStats.cs]: there is already a stats collection that has benchmarks. why new collection is needed?

**Comment by noescape00** [comment]
What is the purpose of those changes? 

Also I think those changes shouldn't be in `NodeStats.cs`.

**Comment by fassadlr** [comment]
@noescape00 OK so the idea here is that instead of returning strings from each component, we return an stats or snapshot object that we can then use to interrogate. This way we can add to the snapshot object more easily for reporting without having to constantly append to the string when we want stats.

**Comment by noescape00** [comment]
```
@noescape00 OK so the idea here is that instead of returning strings from each component, we return an stats or snapshot object that we can then use to interrogate. This way we can add to the snapshot object more easily for reporting without having to constantly append to the string when we want stats.
```

Right now all 3 types of stats are used only for logging to the console. 
I think if we need to collect snapshots it shouldn't be done in node stats but instead in a different component. 

`This way we can add to the snapshot object more easily for reporting without having to constantly append to the string when we want stats`

Could you please elaborate on this? I didn't quite understand what you mean

**Comment by dangershony** [comment]
What we should do is have public perf counters on every component that needs display stats.

Then the collection of those stats can happen in an external component without polluting internally any code targeting a specific outputs (like console in this case)

**Comment by dangershony** [comment]
Closing this PR it should be rpelaced with a PR that create a perf counter instead

**Pull request closed without merging**